src/components/Popup/Popup.js

```diff
@@ -23,139 +24,114 @@ import {
   InnerWrapperStyled
 } from './PopupStyled';
 import popupData from './Popup.const';
+import { ReactComponent as WelcomeIcon } from './images/welcome.svg';
+import { ReactComponent as ConsentsIcon } from './images/icon_terms.svg';
 
-class Popup extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      step: 1,
-      updatedConsents: [],
-      isLoading: false,
-      allowSubmitConsents: false
-    };
-  }
+const Popup = ({ t }) => {
+  const [step, setStep] = useState(1);
+  const [updatedConsents, setUpdatedConsents] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [allowSubmitConsents, setAllowSubmitConsents] = useState(false);
+  const { popupType, consents } = useSelector(state => state.popup);
+  const dispatch = useDispatch();
 
-  componentDidMount() {
-    const { consents } = this.props;
-    this.setState({
-      updatedConsents: consents
-    });
-    this.checkAccess(consents);
-  }
-
-  renderNextStep = () => {
-    this.setState(prevState => {
-      return { step: prevState.step + 1 };
-    });
+  const checkAccess = items => {
+    const notCheckedTerm = items.find(
```

Comment:
we could rename it to more meaningful, eg. `shouldBlockConfirmButton` / `requiredConsentsToAccept` or something similar